test/e2e/cypress/integration/profiles/oidc-provider/mfa.spec.ts

```diff
@@ -0,0 +1,242 @@
+                          ),
+                        )
+                        expect(idToken).to.have.property("amr")
+                        expect(idToken.amr).to.deep.equal(["password", "totp"])
+                      })
+                  })
+              })
+
+            // We shouldn't need to authenticate again
+            state = uuid.v4().toString().replace(/-/g, "")
+            nonce = uuid.v4().toString().replace(/-/g, "")
+            scope = ["offline", "openid"]
+            url = oauth2.getAuthorizeURL(
+              client.auth_endpoint,
+              "",
+              client.id,
+              "0",
+              nonce,
+              "https://httpbin.org/anything",
+              "code",
+              ["offline", "openid"],
+              state,
+              undefined,
+            )
+            cy.get("body")
+              .then((body$) => {
+                // Credits https://github.com/suchipi, https://github.com/cypress-io/cypress/issues/944#issuecomment-444312914
+                const appWindow = body$[0].ownerDocument.defaultView
+                const appIframe =
+                  appWindow.parent.document.querySelector("iframe")
+
+                return new Promise((resolve) => {
+                  appIframe.onload = () => resolve(undefined)
+                  appWindow.location.href = url
+                })
+              })
+              .then(() => {
+                // We get the consent screen instead of login
+                cy.get("#openid").click()
+                cy.get("#offline").click()
+                cy.get("#accept").click()
+
+                cy.location("href")
+                  .should(
+                    "match",
+                    new RegExp("https://httpbin.org/anything[?]code=.*"),
+                  )
+                  .then((body) => {
+                    cy.get("body")
+                      .invoke("text")
+                      .then((text) => {
+                        const result = JSON.parse(text)
+                        const tokenParams = {
+                          code: result.args.code,
+                          redirect_uri: "https://httpbin.org/anything",
+                          scope: scope.join(" "),
+                        }
+                        oauth2
+                          .getToken(
+                            client.token_endpoint,
+                            client.id,
+                            client.secret,
+                            "authorization_code",
+                            tokenParams.code,
+                            tokenParams.redirect_uri,
+                            tokenParams.scope,
+                          )
+                          .then((res) => {
+                            const token = res.body
+                            expect(token).to.have.property("access_token")
+                            expect(token).to.have.property("id_token")
```

Comment:
This test looks a bit bloatet, maybe we can improve it by using some common functions to handle the core oidc-provider tasks?

1. Create a valid login URL
2. Exchange the code for a token
3. Accept consent with specific set of scopes

I think that could greatly reduce code complexity and improve readability!